### PR TITLE
CUSTCOM-70 Fixed jersey-client issues with embedded distros of Payara

### PR DESCRIPTION
**Pull request by dmatej**
# Description

# Testing

```
mvn clean install -TC4 -PQuickBuild,fastest
mvn clean install -PBuildExtras,QuickBuild,fastest -pl :payara-embedded-all
mvn clean install -Ppayara-samples -pl :embedded-vs-jersey
```

### Test suites executed

- Quicklook Y
- Payara Samples Y
- Java EE7 Samples N
- Java EE8 Samples N
- Payara Private Tests N
- Payara Microprofile TCKs Runner N
- Jakarta TCKs - jaxrs
- Mojarra N
- Cargo Tracker N

### CODE REVIEW

**Comment by dmatej** [comment]
Jenkins test please

File: appserver/payara-appserver-modules/jaxrs-client-tracing/src/main/java/fish/payara/requesttracing/jaxrs/client/JaxrsClientRequestTracingFilter.java [hunk 1]
```diff
@@ -80,30 +74,13 @@
...
  */
 public class JaxrsClientRequestTracingFilter implements ClientRequestFilter, ClientResponseFilter {
-    public static String REQUEST_CONTEXT_TRACING_PREDICATE = "fish.payara.requesttracing.jaxrs.client.TracingPredicate";
-
-    private ServiceLocator serviceLocator;
-    private RequestTracingService requestTracing;
-    private OpenTracingService openTracing;
+    public static final String REQUEST_CONTEXT_TRACING_PREDICATE = "fish.payara.requesttracing.jaxrs.client.TracingPredicate";
 
-    /**
```
Comment (dmatej) [inline appserver/payara-appserver-modules/jaxrs-client-tracing/src/main/java/fish/payara/requesttracing/jaxrs/client/JaxrsClientRequestTracingFilter.java]: This part caused problems with embedded.

File: appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/OpenTracingApplicationEventListener.java [hunk 2]
```diff
@@ -83,19 +81,10 @@
...
         LOG.finest("postConstruct()");
-        final ServiceLocator serviceLocator = Globals.getDefaultBaseServiceLocator();
-        if (serviceLocator == null) {
-            LOG.config("Default base service locator is null, JAX-RS server tracing is disabled.");
-            return;
-        }
-        final InvocationManager invocationManager = serviceLocator.getService(InvocationManager.class);
-        this.requestTracing = serviceLocator.getService(RequestTracingService.class);
-        this.openTracing = serviceLocator.getService(OpenTracingService.class);
-        if (invocationManager == null || this.openTracing == null) {
```
Comment (dmatej) [inline appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/OpenTracingApplicationEventListener.java]: Do you remember discussion about this if? :-)

File: appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/cdi/TracedInterceptor.java [hunk 3]
```diff
@@ -198,7 +197,7 @@ private String getOperationName(final InvocationContext invocationContext, final
             .getConfigOverrideValue(Traced.class, "operationName", invocationContext, String.class)
             .orElse(traced.operationName());
 
-        if (!"".equals(operationName)) {
+        if (operationName == null || !operationName.isEmpty()) {
```
Comment (pdudits) [inline appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/cdi/TracedInterceptor.java]: While this is faithful translation of original condition, I don't think operationName can be null, as it is result of `Optional.orElse(annotationAttribute with default "")`.

Comment (dmatej) [reply appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/cdi/TracedInterceptor.java]: Fixed, nice catch, I did not notice the Optional :)

**Comment by dmatej** [comment]
Jenkins test please

**Pull request merged**